geopandas/io/parquet.py

```diff
@@ -0,0 +1,285 @@
+    df = DataFrame(df.copy())
+
+    # Encode all geometry columns to WKB
+    for col in geometry_columns:
+        df[col] = to_wkb(df[col].values)
+
+    table = Table.from_pandas(df, preserve_index=index)
+
+    # Store geopandas specific file-level metadata
+    # This must be done AFTER creating the table or it is not persisted
+    metadata = table.schema.metadata
+    metadata.update(
+        {
+            "geo": _encode_metadata(
+                {
+                    "primary_column": geometry_column,
+                    "columns": column_metadata,
+                    "schema_version": METADATA_VERSION,
+                    "creator": {
+                        "library": "geopandas",
+                        "version": geopandas.__version__,
+                    },
+                }
+            )
+        }
+    )
+
+    table = table.replace_schema_metadata(metadata)
+    parquet.write_table(table, path, compression=compression, **kwargs)
+
+
+def read_parquet(path, columns=None, **kwargs):
+    """
+    Load a parquet object from the file path, returning a GeoDataFrame.
+
+    You can read a subset of columns in the file using the ``columns`` parameter.
+    However, the structure of the returned GeoDataFrame will depend on which
+    columns you read:
+    * if no geometry columns are read, this will raise a ``ValueError`` - you
+      should use the pandas `read_parquet` method instead.
+    * if the primary geometry column saved to this file is not included in
+      columns, the first available geometry column will be set as the geometry
+      column of the returned GeoDataFrame.
+
+    Requires 'pyarrow'.
+
+    Parameters
+    ----------
+    path : str, path object or file-like object
+        Any valid string path is acceptable. The string could be a URL. Valid
+        URL schemes include http, ftp, s3, and file. For file URLs, a host is
+        expected. A local file could be:
+        ``file://localhost/path/to/table.parquet``.
+
+        If you want to pass in a path object, geopandas accepts any
+        ``os.PathLike``.
+
+        By file-like object, we refer to objects with a ``read()`` method,
+        such as a file handler (e.g. via builtin ``open`` function)
+        or ``StringIO``.
+    columns : list-like of strings, default=None
+        If not None, only these columns will be read from the file.  If
+        the primary geometry column is not included, the first secondary
+        geometry read from the file will be set as the geometry column
+        of the returned GeoDataFrame.  If no geometry columns are present,
+        a ``ValueError`` will be raised.
+    **kwargs
+        Any additional kwargs are passed to the engine.
+
+    Returns
+    -------
+    GeoDataFrame
+    """
+
+    import_optional_dependency(
+        "pyarrow", extra="pyarrow is required for parquet support."
+    )
+    from pyarrow import parquet
+
+    path, _, _, should_close = get_filepath_or_buffer(path)
```

Comment:
One additional concern here is that `get_filepath_or_buffer` is also not really a public API from pandas .. 

We should maybe check what it does exactly (and which aspects of it we find important). For example, for `read_file` we have some url-handling ourselves (which is *one* of the things that `get_filepath_or_buffer` does):

https://github.com/geopandas/geopandas/blob/f70a66918df086bd0fca37f03e160224f400fe14/geopandas/io/file.py#L18-L34

Reply:
This came from porting the method from `pandas`.  `pyarrow` requires a different method for reading from a buffer than we are using here.  I think non-local files can be added in a later PR, so that we can get this first one out and well-covered by tests.  As such, I dropped this back to just using a string or path object.

Reply:
>  I think non-local files can be added in a later PR

Yes, that sounds good

Reply:
#1382